vllm/v1/engine/core.py

```diff
@@ -67,15 +68,21 @@ def __init__(
 
         # Setup scheduler.
         if isinstance(vllm_config.scheduler_config.scheduler_cls, str):
+            Scheduler = resolve_obj_by_qualname(
+                vllm_config.scheduler_config.scheduler_cls)
+        else:
+            Scheduler = vllm_config.scheduler_config.scheduler_cls
+
+        # This warning can be removed once the V1 Scheduler interface is
+        # finalized and we can maintain support for scheduler classes that
+        # implement it
+        if Scheduler != V1Scheduler:
             logger.warning(
```

Comment:
Better to use "is" instead of "!="?

Reply:
Honestly... I don't know?

I guess there could be a situation where for a different class, `==` could evaluate to True. But I don't know how class equality is implemented, and whether or not that equality would mean the interface and implementation is all the same 🤷 

I can change to `is`

Reply:
My understanding is `!=` is for values and `is` is for memory address, so we usually use `if obj is None` instead of `if obj == None`, because `None` is a singleton.